tests/test_cases_actions.py

```diff
@@ -0,0 +1,309 @@
+from requests import Response
+from requests.exceptions import HTTPError
+from pytest import fixture
+
+from py42.exceptions import Py42BadRequestError
+from py42.exceptions import Py42NotFoundError
+from py42.exceptions import Py42CaseNameExistsError
+from py42.exceptions import Py42UpdateClosedCaseError
+
+from py42.response import Py42Response
+from .conftest import (
+    assert_success,
+    assert_fail,
+    assert_fail_message,
+    create_fake_connector,
+    create_mock_response,
+    assert_successful_single_data,
+    assert_successful_message,
+    assert_successful_summary,
+)
+
+_TEST_NAME = "test_case"
+_TEST_DESCRIPTION = "test_description"
+_TEST_SUBJECT = "test@example.com"
+_TEST_ASSIGNEE = "admin@example.com"
+_TEST_FINDINGS = "some findings: ```code here```"
+_TEST_USER_UID = "1234"
+_TEST_EVENT_ID = "0_12345_abcdef"
+
+_TEST_CASE_RESPONSE = {
+    "number": 2,
+    "name": _TEST_NAME,
+    "createdAt": "2021-01-01T12:00:00.000000Z",
+    "updatedAt": "2021-01-01T12:00:00.000000Z",
+    "description": _TEST_DESCRIPTION,
+    "findings": _TEST_FINDINGS,
+    "subject": _TEST_USER_UID,
+    "subjectUsername": _TEST_SUBJECT,
+    "status": "OPEN",
+    "assignee": _TEST_USER_UID,
+    "assigneeUsername": _TEST_ASSIGNEE,
+    "createdByUserUid": "987654321098765433",
+    "createdByUsername": "admin@example.com",
+    "lastModifiedByUserUid": "987654321098765433",
+    "lastModifiedByUsername": "admin@example.com",
+}
+
+
+@fixture
+def mock_py42_with_case(mocker, mock_py42_client):
+    mock_cases_response = create_mock_response(mocker, _TEST_CASE_RESPONSE)
+    mock_py42_client.cases.create.return_value = Py42Response(mock_cases_response)
+    mock_py42_client.cases.get.return_value = Py42Response(mock_cases_response)
+    mock_py42_client.cases.update.return_value = Py42Response(mock_cases_response)
+    return mock_py42_client
+
+
+@fixture
+def mock_py42_with_cases(mocker, mock_py42_client):
+    response = {"cases": [_TEST_CASE_RESPONSE], "totalCount": 2}
+
+    def gen(*args, **kwargs):
+        yield create_mock_response(mocker, response)
+
+    mock_py42_client.cases.get_all.side_effect = gen
+    return mock_py42_client
+
+
+class TestCode42CasesConnector(object):
+    def test_handle_action_when_creating_case_calls_with_expected_args_and_sets_success_status(
+        self, mocker, mock_py42_with_case
+    ):
+        mock_user_response = mocker.MagicMock(spec=Response)
+        mock_user_response.text = f'{{"users": [{{"userUid": "{_TEST_USER_UID}"}}]}}'
```

Comment:
We do already have a `mock_py42_with_user` fixture, if that's at all helpful
Your `mock_py42_with_case` fixture could use that as a base fixture instead of  `mock_py42_client`